src/services/services.ts

```diff
@@ -3010,21 +3010,30 @@ namespace ts {
             function tryGetGlobalSymbols(): boolean {
                 let objectLikeContainer = tryGetObjectLikeCompletionContainer(contextToken);
                 if (objectLikeContainer) {
-                    // Object literal expression, look up possible property names from contextual type
+                    // We're looking up possible property names from contextual/inferred/declared type.
                     isMemberCompletion = true;
-                    isNewIdentifierLocation = true;
 
                     let typeForObject: Type;
                     let existingMembers: Declaration[];
 
                     if (objectLikeContainer.kind === SyntaxKind.ObjectLiteralExpression) {
+                        // We are completing on contextual types, but may also include properties
+                        // other than those within the declared type.
+                        isNewIdentifierLocation = true;
+
                         typeForObject = typeChecker.getContextualType(<ObjectLiteralExpression>objectLikeContainer);

```

Comment:
For an assignment pattern, you do not get the contextual type. You get the type from the type annotation or the initializer.